File: adventurer/src/main/kotlin/com/guild/kotlin/adventurer/repo/RoleRepository.java
package com.guild.kotlin.adventurer.repo;

import com.guild.kotlin.adventurer.entities.Role;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.Query;
import org.springframework.data.repository.query.Param;
import org.springframework.stereotype.Repository;

@Repository
public interface RoleRepository extends JpaRepository<Role, Long> {

	@Query("FROM Role WHERE name=:name")
	Role findByName(@Param("name") String name);
}

File: adventurer/src/main/kotlin/com/guild/kotlin/adventurer/repo/RoleRepository.kt
package com.guild.kotlin.adventurer.repo

import com.guild.kotlin.adventurer.entities.Role
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.data.jpa.repository.JpaSpecificationExecutor
import org.springframework.data.jpa.repository.Query
import org.springframework.data.repository.query.Param

interface RoleRepository : JpaRepository<Role, Long>, JpaSpecificationExecutor<Role> {

    fun findByName(name: String?): Role
}